nifi-nar-bundles/nifi-site-to-site-reporting-bundle/nifi-site-to-site-reporting-task/src/main/java/org/apache/nifi/reporting/SiteToSiteBulletinReportingTask.java

```diff
@@ -76,10 +81,16 @@
 
     private volatile long lastSentBulletinId = -1L;
 
+    public SiteToSiteBulletinReportingTask() throws IOException {
+        final InputStream schema = getClass().getClassLoader().getResourceAsStream("schema-bulletins.avsc");
+        recordSchema = AvroTypeUtil.createSchema(new Schema.Parser().parse(schema));
+    }
+
     @Override
     protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
         final List<PropertyDescriptor> properties = new ArrayList<>(super.getSupportedPropertyDescriptors());
         properties.add(PLATFORM);
+        properties.add(RECORD_WRITER);
```

Comment:
Is the intent here to allow all S2S Reporting Tasks to use the record writer if they want, or do we want to make it "mandatory" by including it in the properties from the abstract base class?

Reply:
Yeah I asked myself this question... I was not sure if it'd make sense to move it to the abstract base class in case some users implemented custom impl... but that wouldn't break anything (just the addition of an unused property) so it's probably cleaner to have it in the abstract class. What's your feeling?